numpyro/infer/elbo.py

```diff
@@ -311,53 +312,166 @@ class RenyiELBO(ELBO):
+                batch = numpyro.subsample(data, event_dim=0)
+                loc = w_loc * batch
+                scale = jnp.exp(w_scale * batch)
+                numpyro.sample("latent", dist.Normal(loc, scale))
+
+        def create_plates(data):
+            return numpyro.plate("batch", 10000, subsample_size=100)
+
+        elbo = RenyiELBO(num_particles=10, create_plates=create_plates)
+        svi = SVI(model, guide, optax.adam(0.1), elbo)
+
 
     **References:**
 
     1. *Renyi Divergence Variational Inference*, Yingzhen Li, Richard E. Turner
     2. *Importance Weighted Autoencoders*, Yuri Burda, Roger Grosse, Ruslan Salakhutdinov
     """
 
-    def __init__(self, alpha=0, num_particles=2):
+    def __init__(self, alpha=0, num_particles=2, create_plates=None):
         if alpha == 1:
             raise ValueError(
                 "The order alpha should not be equal to 1. Please use ELBO class"
                 "for the case alpha = 1."
             )
         self.alpha = alpha
+        self.create_plates = create_plates
         super().__init__(num_particles=num_particles)
 
+    def _create_plates(self, key, *args, **kwargs):
+        assert self.create_plates is not None
+        with seed(rng_seed=key):
+            plates = self.create_plates(*args, **kwargs)
+        if isinstance(plates, plate):
+            plates = [plates]
+        assert all(
+            isinstance(p, plate) for p in plates
+        ), "create_plates() returned a non-plate"
+        plate_values = {}
+        for p in plates:
+            with p as idx:
+                plate_values[p.name] = idx
+        return plate_values
+
+    def _single_particle_elbo(
+        self, model, guide, param_map, args, kwargs, rng_key, indep_subsample_plates=()
+    ):
+        model_seed, guide_seed = random.split(rng_key)
+        seeded_model = seed(model, model_seed)
+        seeded_guide = seed(guide, guide_seed)
+        model_trace, guide_trace = get_importance_trace(
+            seeded_model, seeded_guide, args, kwargs, param_map
+        )
+        check_model_guide_match(model_trace, guide_trace)
+        _validate_model(model_trace, plate_warning="loose")
+
+        site_plates = {
+            name: {frame for frame in site["cond_indep_stack"]}
+            for name, site in model_trace.items()
+            if site["type"] == "sample"
+        }
+        # We will compute Renyi elbos separately across dimensions
+        # defined in indep_plates. Then the final elbo is the sum
+        # of those independent elbos.
+        if site_plates:
+            indep_plates = set.intersection(*site_plates.values())
+        else:
+            indep_plates = set()
+        indep_plate_scale = 1.0
+        for frame in indep_plates:
+            subsample_size = frame.size
+            size = model_trace[frame.name]["args"][0]
+            if size > subsample_size:
+                # If a subsample plate is not obtained via create_plates, subsample
```

Comment:
why is this exactly?

Reply:
For each particle, the model/guide is seeded using a particular seed, hence subsample indices will be different across particles.

Reply:
i see it seems like this is something we could simplify with special logic for seeding plates...

Reply:
what if `seed` returned extra plate seed info that could be passed around? something like
```
 seeded_model, plate_seeds = seed(model, model_seed, plate_seeds=None)
 seeded_guide, _ = seed(guide, guide_seed, plate_seeds=plate_seeds)
```

Reply:
that's right. there are several ways to assure constant seeds. the question is whether we want to maintain two behaviors. see my other comment